server/app/lib/utils/fusor/vm_launcher.rb

```diff
@@ -0,0 +1,280 @@
+                             "id" => "",
+                             "bootable" => "1",
+                             "preallocate" => "0"
+                           }.with_indifferent_access
+                         }.with_indifferent_access
+                       }.with_indifferent_access
+                     }.with_indifferent_access
+      end
+      
+      def set_osp_attrs
+        image = Image.create( "name" => @profile_name,
+                              "username" => 'root',
+                              "user_data" => 1,
+                              "uuid" => @cr.available_images.find { |hash| @profile_name == hash.name }.id,
+                              "compute_resource_id" => @cr.id,
+                              "operatingsystem_id" => Operatingsystem.find_by_title('RedHat 7.1')['id'],
+                              "architecture_id" => Architecture.find_by_name('x86_64')['id'])         
+        overcloud = {:openstack_auth_url => "http://#{@deployment.openstack_overcloud_address}:5000/v2.0/tokens",
+                     :openstack_username => 'admin', :openstack_tenant => 'admin',
+                     :openstack_api_key  => @deployment.openstack_overcloud_password }
+        keystone  = Fog::Identity::OpenStack.new(overcloud)
+        tenant    = keystone.get_tenants_by_name(@deployment.label).body["tenant"]
+        neutron   = Fog::Network::OpenStack.new(overcloud)
+        nic       = neutron.list_networks.body["networks"].find { |hash| "#{@deployment.label}-net" == hash["name"] }['id']
+
+        @osp_attrs = {"compute_profile_id" => @cp.id,
+                       "compute_resource_id" => @cr.id,
+                       "vm_attrs" => {
+                         "flavor_ref" => "4",
+                         "network" => "#{@deployment.label}-float-net",
+                         "image_ref" => image.find_by_name(@profile_name).uuid,
+                         "security_groups" => "#{@deployment.label}-sec-group",
+                         "nics" => ["", nic],
+                         "tenant_id" => tenant['id']
+                       }.with_indifferent_access
+                      }.with_indifferent_access
+      end
+
+      def set_common_host_attrs
+        @host_attrs = {"name" => @host_name,
+                      "location_id" => Location.find_by_name('Default Location').id,
+                      "environment_id" => Environment.where(:katello_id => "Default_Organization/Library/Fusor_Puppet_Content").first.id,
+                      "organization_id" => @deployment["organization_id"],
+                      "compute_resource_id" => @cr.id,
+                      "enabled" => "1",
+                      "managed" => "1",
+                      "architecture_id" => Architecture.find_by_name(@architecture)['id'],
+                      "operatingsystem_id" => Operatingsystem.find_by_title(@operatingsystem)['id'],
+                      "domain_id" => 1,
+                      "root_pass" => "dog8code",
```

Comment:
We discussed the password is hardcoded for now because we don't have the WebUI setting this just yet.